src/cloud-functions.ts

```diff
@@ -620,6 +622,13 @@ export function optionsToEndpoint(
     'serviceAccount',
     (sa) => sa
   );
+  convertIfPresent(
+    endpoint,
+    options,
+    'secretEnvironmentVariables',
+    'secrets',
+    (secrets) => secrets.map((secret) => ({ secret }))
```

Comment:
This is where the fully qualified names/versions gets weird because we agreed elsewhere that we'd be setting `key` and not `secret` so that we could bind later. I guess we can, in this case, bind both and not bind `secret` in the params case?

Reply:
Ah that's good point.

I've been thinking the opposite - that v1 Functions SDK will receive secret and the CLI will bind the env key using the secret name. With params context, having the option to later bind the secret makes sense. 

Being restrictive here - to set both `key` and `secret` to the secret input is good and letting params be the one that only binds the `secret` makes more sense to me.